linkerd/app/outbound/src/http/detect.rs

```diff
@@ -22,18 +22,17 @@ impl<N> Outbound<N> {
     >
     where
         I: io::AsyncRead + io::AsyncWrite + io::PeerAddr + std::fmt::Debug + Send + Unpin + 'static,
-        N: svc::NewService<NTgt, Service = NSvc> + Clone + Send + 'static,
+        N: svc::NewService<T, Service = NSvc> + Clone + Send + 'static,
         NSvc: svc::Service<io::EitherIo<I, io::PrefixedIo<I>>, Response = ()> + Send + 'static,
         NSvc::Error: Into<Error>,
         NSvc::Future: Send,
-        NTgt: From<T> + 'static,
-        H: svc::NewService<HTgt, Service = HSvc> + Clone + Send + 'static,
+        H: svc::NewService<U, Service = HSvc> + Clone + Send + 'static,
         HSvc: svc::Service<http::Request<http::BoxBody>, Response = http::Response<http::BoxBody>>,
         HSvc: Clone + Send + Sync + Unpin + 'static,
         HSvc::Error: Into<Error>,
         HSvc::Future: Send,
-        HTgt: From<(http::Version, T)> + svc::Param<http::Version> + 'static,
-        T: Param<SkipHttpDetection> + Clone + Send + Sync + 'static,
+        T: Param<Option<Skip>> + Clone + Send + Sync + 'static,
```

Comment:
This change isn't strictly necessary. Can back it out if you'd prefer the other way. @hawkw